### PR TITLE
Trilinos parallel is_non_negative() function

### PR DESCRIPTION
**Pull request by juleoc02**
Changed the is_non_negative() function to work with distributed vectors in Trilinos. Mostly copied from the above all_zero() function that already worked in parallel

### CODE REVIEW

File: source/lac/trilinos_vector.cc [hunk 1]
```diff
@@ -757,36 +757,28 @@ namespace TrilinosWrappers
...
           ++ptr;
         }
 
-      return flag;
+      // in parallel, check that the vector
+      // is zero on _all_ processors.
+      const Epetra_MpiComm *mpi_comm =
+        dynamic_cast<const Epetra_MpiComm *>(&vector->Map().Comm());
+      Assert(mpi_comm != nullptr, ExcInternalError());
+      unsigned int num_negative = Utilities::MPI::sum(flag, mpi_comm->Comm());
```
Comment (drwells) [inline source/lac/trilinos_vector.cc:780]:
```suggestion
      const auto max_n_negative = Utilities::MPI::max(flag, get_mpi_communicator());
```
its conceivable that a summation could wrap-around for very large vectors, so lets just use max.

**Review by drwells — CHANGES_REQUESTED** [review]
Looks good - I have two small suggestions to make w.r.t. the MPI reduction.

File: source/lac/trilinos_vector.cc (hunk 1 above)
Comment (juleoc02) [reply source/lac/trilinos_vector.cc:780]: This makes a lot of sense.

File: source/lac/trilinos_vector.cc [hunk 2]
```diff
@@ -757,36 +757,25 @@ namespace TrilinosWrappers
...
               break;
             }
           ++ptr;
         }
 
-      return flag;
+      // in parallel, check that the vector
+      // is zero on _all_ processors.
+      const auto max_n_negative = Utilities::MPI::max(flag, get_mpi_communicator());
+      return num_negative == 0;
```
Comment (drwells) [inline source/lac/trilinos_vector.cc:778]: I think this should be `max_n_negative`.

Comment (juleoc02) [reply source/lac/trilinos_vector.cc]: yes it should. :)

**Comment by jppelteret** [comment]
@drwells, @bangerth Do you have any further comments, or is this now ready to be merged?

**Review by bangerth — COMMENTED** [review]
Looks good to me, but @juleoc02 is my student and when possible I try to avoid merging my students' PRs. So I would appreciate someone else do it!

**Review by masterleinad — APPROVED** [review]
Looks good to me. The implementation is pretty much the same as for `all_zero`.

**Review by tamiko — APPROVED** [review]

**Pull request merged**